provisioning_client/tools/RIoT_config/RIoT_config.h

```diff
@@ -0,0 +1,2135 @@
+/**
+ * \file config.h
+ *
+ * \brief Configuration options (set of defines)
+ *
+ *  This set of compile-time options may be used to enable
+ *  or disable features selectively, and reduce the global
+ *  memory footprint.
+ */
+/*
+ *  Copyright (C) 2006-2018, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+
+#ifndef MBEDTLS_CONFIG_H
```

Comment:
This #define is not in-line with the file-name and shows this is not really our source-code.
As far as I can tell, this is taken from https://tls.mbed.org/api/config_8h.html. We should not have a copy in our repo: instead, we can document and maybe have our own include containing the recommended definitions.

Reply:
https://github.com/microsoft/RIoT/blob/master/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/config.h here is where the config.h is actually pulled from, there is a lot of stuff in the RIoT crypto that pulls from that header file, this might need to be explained by the MSR team if we have legal access to this or not

Reply:
@ttins let's reference it from the submodule in that case. No need to create a copy.

Reply:
The problem is in the submodule RIOTBUILD is not defined, its a define we need to add according to @DMattoon, he might be able to chime in on a better way of handling this, I agree that the solution of copy/paste is not elegant

Reply:
It is part of the build parameters/makefiles (e.g., -DRIOTBUILD).  No, it doesn't appear in a header.

Reply:
Sweet so just like we do with curve makefile parameter, that will definitely cleanup some nasty stuff

Reply:
Fixed, much cleaner, thanks @DMattoon